demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+
+            // re-render the component, manage the 'reset' state by yourself
+        };
+        facade.showValidationUI = () => {
+            // TODO: requires implementation
+            /**
+             * The purpose of this method is to update your custom question's UI with visual feedback 
+             * after the learner presses the Check Answer button.
+             * 
+             * The following is an example implementation that shows the standard Learnosity "checkmark / tick" for a correct answer
+             * or the standard Learnosity "x mark / cross" for an incorrect answer. 
+             */
+            
+            // EXAMPLE implemetation:
+
+            const answerIsCorrect = facade.isValid();
+            /**
+             * update the UI based on the result
+             * IMPORTANT: In both cases we need to target the element with the classname lrn_response_input
+             */
+            if (answerIsCorrect) {
+                /** if the answer is correct we will add the Learnosity utility class lrn_correct, which will automatically show the Learnosity
+                 * checkmark (tick) you are used to on regular Learnosity quesiton types
+                 */
+                el.querySelector(".lrn_response_input").classList.add("lrn_correct");
+            } else {
+                /** if the answer is incorrect we will add the Learnosity utility class lrn_incorrect, which will automatically show the Learnosity
+                 * x (cross) you are used to on regular Learnosity quesiton types
+                 */
+                el.querySelector(".lrn_response_input").classList.add("lrn_incorrect");
+            }
+                          
+        };
+        facade.resetValidationUI = () => {
+            // TODO: requires implementation
+            /**
+             * If you implement showValidationUI() above, then you need to implement this method also.
+             * 
+             * If the user enters a response to your custom question, presses the Check Answer button, and then changes their response to something new,
+             * you now need to remove the feedback UI that you had displayed for the previous response. This is exactly the purpose of this method.
+             * 
+             * The following is an example implementation that removes the standard Learnosity validation UI ("checkmark / tick" if the previous response was correct
+             * or "x mark / cross" if the previous response was incorrect).
+             */
+
+            // Example implementation:
+
+            // // we will remove the Learnosity checkmark or x previously showing
+            el.querySelector(".lrn_response_input").classList.remove("lrn_correct");
+            el.querySelector(".lrn_response_input").classList.remove("lrn_incorrect");
+            
+            // // if we chose to also display the correct answer, we will now remove the display of the correct answer as well by 
```

Comment:
Fix comments for readability
```suggestion
            // if we chose to also display the correct answer,
            // we will now remove the display of the correct answer as well by 
```